engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/transform/query/NadelQueryTransformer.kt

```diff
@@ -137,6 +115,29 @@ class NadelQueryTransformer internal constructor(
         return patchedArtificialFields + patchedNewField
     }
 
+    private suspend fun applyTransformationSteps(
+        field: ExecutableNormalizedField,
+        transformationSteps: List<NadelExecutionPlan.Step<Any>>,
+        executionContext: NadelExecutionContext,
+        continuation: Continuation,
+        service: Service
+    ): NadelTransformFieldResult {
+        var fieldFromPreviousTransform: ExecutableNormalizedField = field
+        var transformFieldResult: NadelTransformFieldResult? = null
```

Comment:
We need to merge these results instead of just replacing them.

We use these results to keep track of artificial fields which then get automatically removed at the very end of all the transformations.